### PR TITLE
DHFPROD-7402:Ensure related entity mapping id is unique

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
Ensure related entity mapping id is unique
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/mappable-entity-lib.sjs [hunk 1]
```diff
@@ -35,32 +35,46 @@ function addRelatedMappableEntities(entityName, targetEntityName, entityProperti
...
       let joinPropertyName = entityPropertyValue["joinPropertyName"] ? entityPropertyValue["joinPropertyName"] : entityPropertyValue.items["joinPropertyName"]
       if(relatedEntityName==targetEntityName){
-        const entityMappingId = entityName + ":" + targetEntityName + "." + joinPropertyName;
-        const mappingTitle = entityName + " (" + entityPropertyName + " "+ relatedEntityName + ")";
-        entityResponseObject.entityMappingId = entityMappingId;
-        entityResponseObject.mappingTitle = mappingTitle;
-        setRelatedEntityMappings(responseArray.find(response => response.entityType == targetEntityName), mappingTitle, entityMappingId);
+        const targetEntityResponse = responseArray.find(response => response.entityType == targetEntityName);
+        let relatedEntityMapping;
+        if(targetEntityResponse.relatedEntityMappings){
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/mappable-entity-lib.sjs:40]: This looks like it should be a separate function - the function is already 30 lines long with 6 arguments, which is pretty complex - VSCode's sonarlint plugin gives it a complexity score of 21 (the plugin defaults to a max score of 15, so this gets flagged). 

You may also want to combine some of the arguments into an object with a self-describing name, such as "entityContext".

**Review by rjrudin — CHANGES_REQUESTED** [review]
Looks good, the library module just needs some refactoring to simplify the function.

**Comment by srinathgit** [comment]
This PR needs some UI change as well which @briantang will make once middle-tier portion of the work . It shouldn't be merged unless the UI changes are included as well as it will break the mapping UI. So, I am unchecking some of the checklist so that the PR is not merged

**Review by rjrudin — APPROVED** [review]

**Comment by srinathgit** [comment]
@briantang 
UI changes seem good to me. I don't see any obvious issues.

**Review by rjrudin — APPROVED** [review]

**Comment by briantang** [comment]
Pushed an RTL unit test fix

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/mappable-entity-lib.sjs (hunk 1 above)
Comment (wooldridge) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/mappable-entity-lib.sjs]: @rjrudin sonarlint sounds cool, I'll have to check it out.

**Review by wooldridge — APPROVED** [review]

**Review by briantang — APPROVED** [review]

**Pull request merged**